### DISCUSSION TITLE
Background for source code and horizontal lines

### DISCUSSION THREAD

**Discussion by lukas-reineke** [nvim-orgmode/orgmode#83] [Ideas]
I have a function in my dotfiles that I originally wrote for markdown, that adds background to code blocks, and horizontal lines for `-----`.
The code is here https://github.com/lukas-reineke/dotfiles/blob/master/vim/lua/org.lua

![Screen Capture_select-area_20210909151830](https://user-images.githubusercontent.com/12900252/132633362-fce0638a-fa47-4563-9707-66875b21a1ff.png)

Are you interested in adding this here? I could clean this up a bit and make a PR.

**Comment by kristijanhusak** [comment]
This looks great, but I'm trying to have only features that exists in emacs orgmode.
There is a horizontal rule in the emacs orgmode (https://orgmode.org/manual/Horizontal-Rules.html), but there's no way to present it like this there. It only generates horizontal rule on export.
If you think this will be useful for more people, I advise to implement it as a separate plugin. I'll gladly add it to the readme so other people can find it.

Also, for the code block highlighting, you can achieve similar effect if you are using TS highlights by creating `after/queries/org/highlights.scm` in your dots and adding this:
```
(block) @markdownCodeBlockBG
```
It doesn't highlight the whole line though, only the content length, which is probably not the desired effect. Here's example with my colorscheme using `Search` hl:

![block](https://i.imgur.com/G0T686I.png)

**Reply by lukas-reineke** [reply]
Yeah, thinking about this more, making it a separate plugin makes more sense.
This way I can also combine markdown and whatever other filetype into the same code.
And it is just eyecandy, a lot of people would probably consider this bloat.

> Also, for the code block highlighting, you can achieve similar effect if you are using TS highlights by creating after/queries/org/highlights.scm in your dots and adding this:

I didn't know you can add queries like this. That's cool! thanks.
But yeah, the advantage of using marks is that it highlights the whole line.

I'll throw the code in a new plugin and update here once it's done.

**Comment by lukas-reineke** [comment]
Here is the plugin

https://github.com/lukas-reineke/headlines.nvim

**Reply by kristijanhusak** [reply]
Oh nice, that was fast :D

**Reply by kristijanhusak** [reply]
can you add "orgmode-nvim" to the topics? I just added a note [here](https://github.com/kristijanhusak/orgmode.nvim#plugins) how to list all available plugins via topic so it would be great to have it.
